packages/cspell-lib/src/lib/Document/isBinaryDoc.ts

```diff
@@ -1,19 +1,22 @@
-import { isGenerated, isGeneratedFile } from '../LanguageIds.js';
+import { getLanguagesForBasename, isGenerated } from '../LanguageIds.js';
 import type { Uri } from '../util/Uri.js';
 import { basename, toUri } from '../util/Uri.js';
 import type { Document } from './Document.js';
 import { normalizeLanguageIds } from './normalizeLanguageIds.js';
 
 export function isBinaryDoc(document: Document): boolean {
-    return isBinaryFile(toUri(document.uri), document.languageId);
+    return isBinaryFile(toUri(document.uri), document.languageId, document.text);
 }
 
-export function isBinaryFile(filename: Uri | URL | string, languageId?: string | string[]): boolean {
+export function isBinaryFile(filename: Uri | URL | string, languageId?: string | string[], text?: string): boolean {
     const filenameUri = toUri(filename);
     if (languageId) {
         const ids = normalizeLanguageIds(languageId);
         if (ids.length) return isGenerated(ids);
     }
     const file = basename(filenameUri);
-    return isGeneratedFile(file);
+    const ids = getLanguagesForBasename(file);
+    if (ids.length) return isGenerated(ids);
+    // Unknown file type, check the content.
+    return text?.slice(0, 1024).includes('\u0000') || false;
```

Comment:
so you are looking for `\0` to check if file is a binary ? how can be sure it's statiscally correct ?

maybe a lib like this ?

https://github.com/bevry/istextorbinary
https://github.com/gjtorikian/isBinaryFile

Reply:
@ccoVeille,

Thank you for taking a look. In this case we are only looking at the contents of an unknown file type. Since the file loader already converts UTF16 to text, the occurrence of `\0` should be minimal (since `\0` doesn't occur is most text files). 

In any case, I'll take a look at the suggested libraries.

Reply:
@ccoVeille,

- I took a look at [isBinaryFile](https://github.com/gjtorikian/isBinaryFile). It mostly does exactly the same thing, searching for `\0` or `null` as they call it. It also checks for weird character combinations.

- [istextorbinary](https://github.com/bevry/istextorbinary) does look at the file name to and then at the content (exactly like above), but it only checks to see if the data is UTF8. It does seem to include two other packages `textextensions ` and `binaryextensions`. I'll take a look to see if their lists are more extensive.

Reply:
Thanks for checking. I think you code is robust enough then?

Reply:
I think it is fine. Especially since it only runs when it is an unknown file type.